homeassistant/components/climate/generic_thermostat.py

```diff
@@ -222,7 +285,26 @@ def current_operation(self):
     @property
     def target_temperature(self):
         """Return the temperature we try to reach."""
-        return self._target_temp
+        if STATE_AUTO in self.operation_list:
```

Comment:
Store auto mode differently. It's inefficient to check if an item exists in a list 3 times on each update.